src/ngraph/op/fused/batch_to_space.cpp

```diff
@@ -0,0 +1,208 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <cmath>
+#include <cstddef>
+#include <memory>
+#include <ops.hpp>
+
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/fused/batch_to_space.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v1::BatchToSpace::type_info;
+
+ngraph::op::v1::BatchToSpace::BatchToSpace(const ngraph::Output<ngraph::Node>& data,
+                                           const ngraph::Output<ngraph::Node>& block_shape,
+                                           const ngraph::Output<ngraph::Node>& crops_begin,
+                                           const ngraph::Output<ngraph::Node>& crops_end)
+    : FusedOp({data, block_shape, crops_begin, crops_end})
+{
+    constructor_validate_and_infer_types();
+}
+
+NodeVector op::v1::BatchToSpace::decompose_op() const
+{
+    auto data = input_value(0);
+    auto block = input_value(1);
+    auto crops_begin = input_value(2);
+    auto crops_end = input_value(3);
+
+    const auto& data_shape = data.get_shape();
+
+    NODE_VALIDATION_CHECK(this,
+                          (data_shape.size() >= 2),
+                          "The data tensor with rank lower than 2 is not supported (data rank: ",
+                          data_shape.size(),
+                          ")");
+
+    const auto block_const = as_type_ptr<op::Constant>(block.get_node_shared_ptr());
+    const auto crops_begin_const = as_type_ptr<op::Constant>(crops_begin.get_node_shared_ptr());
+    const auto crops_end_const = as_type_ptr<op::Constant>(crops_end.get_node_shared_ptr());
+
+    auto get_in_vec_int64 = [](const shared_ptr<op::Constant>& in_const, vector<int64_t>& values) {
+        if (in_const->get_element_type() == element::i32)
+        {
+            auto tmp = in_const->get_vector<int32_t>();
+            values.insert(values.begin(), tmp.begin(), tmp.end());
+        }
+        else
```

Comment:
Don't assume `int32` and `int64` cover all cases; give an error in case the checks in validate change later.

Reply:
I think that we can use `cast_vector<int64_t>` in this case, look at #4191

Reply:
fixed